src/models.ts

```diff
@@ -32,38 +32,48 @@ export class QueryResult {
     }
 
     static async run(client : QueryApi, query : string) : Promise<QueryResult> {
-        const result = new QueryResult()
-        let currentSchema : FluxTableMetaData
-        let currentTableResult : TableResult
         return new Promise((resolve, reject) => {
-            client.queryRows(query, {
-                next(row : string[], tableMeta : FluxTableMetaData) {
-                    if (currentSchema !== tableMeta) {
-                        if (currentTableResult !== undefined) {
-                            // "Complete" the table by pushing it on to the result list before creating a new
-                            // table.
-                            result.push(currentTableResult)
-                        }
-                        // Copy the columns so there isn't a circular dependency.
-                        const columns = tableMeta.columns.map(x => Object.assign({}, x))
-                        currentTableResult = new TableResult(columns, row)
-                        currentSchema = tableMeta
-                    } else {
-                        currentTableResult.push(row)
-                    }
-                },
-                error(error : Error) {
-                    reject(error)
-                },
-                complete() {
-                    // The last row doesn't know it's the last row, so
-                    // add the last table result, if there are any.
-                    if (currentTableResult !== undefined) {
-                        result.push(currentTableResult)
-                    }
-                    resolve(result)
-                }
-            })
+            client.queryRows(query, queryResultObserver(resolve, reject))
+        })
+    }
+
+    static async parseCSVResponse(response : AnnotatedCSVResponse) : Promise<QueryResult> {
```

Comment:
Hm, I don't quite understand the need for this. The intent of the upstream changes to the client apis was that it had the same api as the run `queryRows` api, so would technically only need to use the `queryResultObserver` - if that's not the case, I think we need to file a bug.

If we can't get that same API, I think this function (a) should move to where it's used, and (b) should have a comment with the link to the bug about having an inconsistent API.

Reply:
Query has `client.queryRows(query, queryResultObserver(resolve, reject))` and invoke uses `client.invoke(id).consumeRows(queryResultObserver(resolve, reject))`.

Do you mean that it should be `client.invoke(id, queryResultObserver(resolve, reject))`? I tweaked it a bit in the latest https://github.com/influxdata/vsflux/pull/359/commits/564322f67e234bf39410c421e6a52180c2262254

Reply:
Hm, that feels inconsistent, but nothing we can do about it, I guess.